addons/website_discuss_room/static/src/js/chat_room.js

```diff
@@ -0,0 +1,127 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { renderToElement } from "@web/core/utils/render";
+import { rpc } from "@web/core/network/rpc";
+
+publicWidget.registry.ChatRoom = publicWidget.Widget.extend({
+    selector: '.o_wdiscuss_room_widget',
+    events: {
+        'click .o_wdiscuss_room_link': '_onChatRoomClick',
+    },
+
+    /**
+      * Manage the chat room, update the participant count...
+      *
+      * The widget takes some options
+      * - 'room-name', the name of the discuss room
+      * - 'chat-room-id', the ID of the `chat.room` record
+      * - 'auto-open', the chat room will be automatically opened when the page is loaded
+      * - 'check-full', check if the chat room is full before joining
+      * - 'attach-to', a JQuery selector of the element on which we will add the discuss
+      *                iframe. If nothing is specified, it will open a modal instead.
+      * - 'default-username': the username to use in the chat room
+      * - 'room-server': the domain name of the Chat room server to use
+      */
+    start: async function () {
+        await this._super.apply(this, arguments);
+        this.roomName = this.$el.data('room-name');
```

Comment:
While doing this, we could perhaps remove the jQuery ? :disguised_face:

Reply:
I should have remove all of it but if you find a survivor tell me ^^